plugins/mcumgr/smp_group_custom.cpp

```diff
@@ -0,0 +1,265 @@
+/******************************************************************************
+** Copyright (C) 2023 Jamie M.
+**
+** Project: AuTerm
+**
+** Module:  smp_group_custom.cpp
+**
+** Notes:
+**
+** License: This program is free software: you can redistribute it and/or
+**          modify it under the terms of the GNU General Public License as
+**          published by the Free Software Foundation, version 3.
+**
+**          This program is distributed in the hope that it will be useful,
+**          but WITHOUT ANY WARRANTY; without even the implied warranty of
+**          MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+**          GNU General Public License for more details.
+**
+**          You should have received a copy of the GNU General Public License
+**          along with this program.  If not, see http://www.gnu.org/licenses/
+**
+*******************************************************************************/
+#include "smp_group_custom.h"
+
+enum modes : uint8_t {
+    MODE_IDLE = 0,
+    MODE_COMMAND,
+};
+
+
+static QStringList smp_error_defines = QStringList() <<
+    //Error index starts from 2 (no error and unknown error are common and handled in the base code)
+    "INVALID_FORMAT" <<
+    "QUERY_YIELDS_NO_ANSWER";
+
+static QStringList smp_error_values = QStringList() <<
+    //Error index starts from 2 (no error and unknown error are common and handled in the base code)
+    "The provided format value is not valid" <<
+    "Query was not recognized";
```

Comment:
This will not work, this is for error codes that are received from the SMP server, e.g. the img group errors are here in zephyr: https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/mgmt/mcumgr/grp/img_mgmt/img_mgmt.h#L63